numpy/lib/tests/test_function_base.py

```diff
@@ -3432,6 +3432,16 @@ def mean(self, axis=None, dtype=None, out=None):
         a = MySubClass([1, 2, 3])
         assert_equal(np.median(a), -7)
 
+    @pytest.mark.parametrize('arr',
+                             ([1., 2., 3.], [1., np.nan, 3.], np.nan, 0.))
```

Comment:
These tests failed for the cases that include `np.nan`

Reply:
By the looks of the array-subclass tests are also missing from the `nanmedian` test suite.
Might it be an idea to add this new test there as well?

Reply:
@BvB93 - indeed, that would have been good... and indeed even with this PR the following fails:
```
In [2]: np.nanmedian([np.nan, np.nan] << u.km)
/usr/lib/python3/dist-packages/numpy/lib/nanfunctions.py:1113: RuntimeWarning: All-NaN slice encountered
  r, k = function_base._ureduce(a, func=_nanmedian, axis=axis, out=out,
Out[2]: nan
```

Reply:
Oops, I checked badly - with this PR, `nanmedian` is in fact correct too; it is just missing a test.